pandas/core/indexes/base.py

```diff
@@ -5155,6 +5155,9 @@ def _get_engine_target(self) -> ArrayLike:
         if isinstance(vals, StringArray):
             # GH#45652 much more performant than ExtensionEngine
             return vals._ndarray
+        if type(self) is Index and isinstance(self._values, ExtensionArray):
+            # TODO(ExtensionIndex): remove special-case, just use self._values
+            return self._values.astype(object)
```

Comment:
do we need to do something with _engine_type?

Reply:
The `Index._engine_type` is still hardcoded to be `ObjectEngine`, regardless of the values it stores. It's only in the `_engine` property we switch to ExtensionEngine based on the return value of `_get_engine_target`.